### PR TITLE
reimplements the family manifest modal

### PR DESCRIPTION
**Pull request by hlminh2000**

### CODE REVIEW

File: src/components/DownloadManifestModal.js [hunk 1]
```diff
@@ -30,103 +29,153 @@ const Button = compose(withTheme)(({ theme, children, ...props }) => (
...
-                    height: 15,
-                    marginRight: 9,
-                  }}
-                />
-              )}
-              <Trans>Generate KF-get ID</Trans>
-            </Button>
+            <Fragment>
+              <span ref={copyRef} className={`copyContent`}>
+                {setId || <Trans>Generate KF-get ID</Trans>}
```
Comment (jgnieuwhof) [inline src/components/DownloadManifestModal.js]: Is KF-get still the name for this? Thought the design may have included something different

File: src/components/FamilyManifestModal/FamilyDataTypesStatsQuery.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return {
+          ...acc,
+          [`sqon${i}`]: {
+            op: 'and',
+            content: dataTypeFilters(participantIds),
+          },
+        };
+      }, {})}
+      render={({ data, loading }) => {
+        return children(
```
Comment (jgnieuwhof) [inline src/components/FamilyManifestModal/FamilyDataTypesStatsQuery.js]: not necessary, but maybe for consistency `render` may be preferred over `children` ?

File: src/components/FamilyManifestModal/queries.js [hunk 3]
```diff
@@ -81,5 +113,18 @@ export const generateFamilyManifestModalProps = async ({ api, sqon }) => {
     api,
     familyMemberIds: familyMembersWithoutParticipantIds,
   });
-  return { familyMemberIds, participantIds, familyMembersWithoutParticipantIds, dataTypes };
+  const { participantFilesCount, participantFilesSize } = await participantsFilesCountAndSize({
+    api,
+    sqon,
+  });
+  console.log('participantFilesCount: ', participantFilesCount);
+  console.log('participantFilesSize: ', participantFilesSize);
```
Comment (jgnieuwhof) [inline src/components/FamilyManifestModal/queries.js]: :fire:

File: src/components/DownloadManifestModal.js (hunk 1 above)
Comment (hlminh2000) [reply src/components/DownloadManifestModal.js]: very nice catch! indeed it says `Generate manifest ID` :)

File: src/components/FamilyManifestModal/FamilyDataTypesStatsQuery.js (hunk 2 above)
Comment (hlminh2000) [reply src/components/FamilyManifestModal/FamilyDataTypesStatsQuery.js]: addressed! :)

File: src/components/DownloadManifestModal.js [hunk 4]
```diff
@@ -30,103 +29,155 @@ const Button = compose(withTheme)(({ theme, children, ...props }) => (
...
+              const type = 'file';
+              const { data, errors } = await saveSet({
+                type,
+                sqon: sqon || {},
+                userId: loggedInUser.egoId,
+                path: 'kf_id',
+                api: graphql(api),
+              });
+              if (errors && errors.length) {
+                setWarning('Unable to generate KF-get ID, please try again later.');
```
Comment (jgnieuwhof) [inline src/components/DownloadManifestModal.js]: KF-get

File: src/components/FamilyManifestModal/FamilyDataTypesStatsQuery.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+          [`sqon${i}`]: {
+            op: 'and',
+            content: dataTypeFilters(participantIds),
+          },
+        };
+      }, {})}
+      render={({ data, loading }) => {
+        return render(
+          loading
+            ? { loading }
```
Comment (jgnieuwhof) [inline src/components/FamilyManifestModal/FamilyDataTypesStatsQuery.js]: might be helpful to return a default data empty array here to guard against null exceptions for consumers?
```
return render({ loading, data: loading ? [] : dataTypes.map(...) })
```
?

File: src/components/FamilyManifestModal/queries.js [hunk 6]
```diff
@@ -1,6 +1,38 @@
...
+          }
+        }
+      `,
+    variables: { sqon, include_missing: true },
+  });
+
+  const { data } = response;
+
+  const participantFilesCount = get(data, 'file.hits.total') || 0;
+  const participantFilesSize = get(data, 'file.aggregations.size.stats.sum') || 0;
```
Comment (jgnieuwhof) [inline src/components/FamilyManifestModal/queries.js]: I wouldn't bother unpacking `{ data }` above if using `get`, also get defaults = third arg
```
return { participantFilesCount: get(data, 'data.file.hits.total', 0), ... }
```
also with that change I wouldn't include the intermittent vars, just construct the json in the return statement

**Pull request merged**